### PR TITLE
Add priority peer option (for AA needs)

### PR DESCRIPTION
**Pull request by marcindsobczak**
## Changes:
- add IsPriority option in SyncPeer and ProtocolHandlerBase (false as default)
- add PriorityPeerMaxCount to network config (0 as default)
- don't drop priority peers because of lowest head number as long as PriorityPeerCount < PriorityPeerMaxCount
- add PriorityPeers metric
- add AaPriorityPeersMaxCount to AA config with default value of 20. This value is ~~overwritting~~increasing  PriorityPeersMaxCount in network config

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

We should run AA nodes from this branch and test their behavior before merging this PR

### CODE REVIEW

**Review by kristofgazso — APPROVED** [review]

File: src/Nethermind/Nethermind.AccountAbstraction/AccountAbstractionPlugin.cs [hunk 1]
```diff
@@ -186,6 +187,7 @@ public Task Init(INethermindApi nethermindApi)
 
             if (_accountAbstractionConfig.Enabled)
             {
+                _nethermindApi.Config<INetworkConfig>().PriorityPeersMaxCount = _accountAbstractionConfig.AaPriorityPeersMaxCount;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.AccountAbstraction/AccountAbstractionPlugin.cs:190]: This seems weird, why those values are connected?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.AccountAbstraction/AccountAbstractionPlugin.cs:190]: Also shouldn't be += ?

File: src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs [hunk 2]
```diff
@@ -225,8 +238,10 @@ internal IEnumerable<SyncPeerAllocation> ReplaceableAllocations
         }
 
         public int PeerCount => _peers.Count;
+        public int PriorityPeerCount => _peers.Values.Count(p => p.SyncPeer.IsPriority.Equals(true));
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs:241]: If used often maybe should be separate counter?

**Review by LukaszRozmej — APPROVED** [review]
Merge after 1.12.7 is released

**Comment by LukaszRozmej** [comment]
Please resolve conflicts

File: src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs (hunk 2 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs:241]: Except of logger in debug mode, it's used only in one place, only if there are priority peers enabled. Adding separate counter will need adding plenty of increments/decrements with checks (if peer is priority then increment/decrement). Not worth imo

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs:241]: I think peers are added only in AddPeer and removed only in RemovePeer, that's not that many checks. and with Interlocked you would have threadsafety

File: src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public abstract class SyncPeerProtocolHandlerBase : ProtocolHandlerBase, ISyncPe
 
         // this means that we know what the number, hash, and total diff of the head block is
         public bool IsInitialized { get; set; }
-        public bool IsPriority { get; set; }
+        public new bool IsPriority { get; set; }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/SyncPeerProtocolHandlerBase.cs:61]: why new?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**